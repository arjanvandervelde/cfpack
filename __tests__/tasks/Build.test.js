const path = require('path');
const BuildTask = require('../../src/tasks/Build');

describe('BuildTask', () => {
	const templates = [
		'and.yml',
		'equals.yml',
		'if.yml',
		'not.yml',
		'or.yml',
	];

	test.each(templates)('::processTemplate --> {%s}', (template) => {
		const task = new BuildTask();
		const filename = path.resolve(__dirname, '../templates/functions/', template);
		const result = task.processTemplate(filename);
		expect(JSON.stringify(result, '', 2)).toMatchSnapshot();
	});
});
