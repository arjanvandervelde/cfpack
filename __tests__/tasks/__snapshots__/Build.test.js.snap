// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildTask ::processTemplate --> {and.yml} 1`] = `
"{
  \\"MyAndCondition\\": {
    \\"Fn::And\\": [
      {
        \\"Fn::Equals\\": [
          \\"sg-mysggroup\\",
          {
            \\"Ref\\": \\"ASecurityGroup\\"
          }
        ]
      },
      {
        \\"Condition\\": \\"SomeOtherCondition\\"
      }
    ]
  }
}"
`;

exports[`BuildTask ::processTemplate --> {equals.yml} 1`] = `
"{
  \\"UseProdCondition\\": {
    \\"Fn::Equals\\": [
      {
        \\"Ref\\": \\"EnvironmentType\\"
      },
      \\"prod\\"
    ]
  }
}"
`;

exports[`BuildTask ::processTemplate --> {if.yml} 1`] = `
"{
  \\"SecurityGroups\\": [
    {
      \\"Fn::If\\": [
        \\"CreateNewSecurityGroup\\",
        {
          \\"Ref\\": \\"NewSecurityGroup\\"
        },
        {
          \\"Ref\\": \\"ExistingSecurityGroup\\"
        }
      ]
    }
  ],
  \\"Outputs\\": {
    \\"SecurityGroupId\\": {
      \\"Description\\": \\"Group ID of the security group used.\\",
      \\"Value\\": {
        \\"Fn::If\\": [
          \\"CreateNewSecurityGroup\\",
          {
            \\"Ref\\": \\"NewSecurityGroup\\"
          },
          {
            \\"Ref\\": \\"ExistingSecurityGroup\\"
          }
        ]
      }
    }
  },
  \\"MyDB\\": {
    \\"Type\\": \\"AWS::RDS::DBInstance\\",
    \\"Properties\\": {
      \\"AllocatedStorage\\": 5,
      \\"DBInstanceClass\\": \\"db.m1.small\\",
      \\"Engine\\": \\"MySQL\\",
      \\"EngineVersion\\": 5.5,
      \\"MasterUsername\\": {
        \\"Ref\\": \\"DBUser\\"
      },
      \\"MasterUserPassword\\": {
        \\"Ref\\": \\"DBPassword\\"
      },
      \\"DBParameterGroupName\\": {
        \\"Ref\\": \\"MyRDSParamGroup\\"
      },
      \\"DBSnapshotIdentifier\\": {
        \\"Fn::If\\": [
          \\"UseDBSnapshot\\",
          {
            \\"Ref\\": \\"DBSnapshotName\\"
          },
          {
            \\"Ref\\": \\"AWS::NoValue\\"
          }
        ]
      }
    }
  },
  \\"UpdatePolicy\\": {
    \\"AutoScalingRollingUpdate\\": {
      \\"Fn::If\\": [
        \\"RollingUpdates\\",
        {
          \\"MaxBatchSize\\": 2,
          \\"MinInstancesInService\\": 2,
          \\"PauseTime\\": \\"PT0M30S\\"
        },
        {
          \\"Ref\\": \\"AWS::NoValue\\"
        }
      ]
    }
  }
}"
`;

exports[`BuildTask ::processTemplate --> {not.yml} 1`] = `
"{
  \\"MyNotCondition\\": {
    \\"Fn::Not\\": [
      {
        \\"Fn::Equals\\": [
          {
            \\"Ref\\": \\"EnvironmentType\\"
          },
          \\"prod\\"
        ]
      }
    ]
  }
}"
`;

exports[`BuildTask ::processTemplate --> {or.yml} 1`] = `
"{
  \\"MyOrCondition\\": {
    \\"Fn::Or\\": [
      {
        \\"Fn::Equals\\": [
          \\"sg-mysggroup\\",
          {
            \\"Ref\\": \\"ASecurityGroup\\"
          }
        ]
      },
      {
        \\"Condition\\": \\"SomeOtherCondition\\"
      }
    ]
  }
}"
`;
